packages/ipfs/src/core/components/add.js

```diff
@@ -0,0 +1,9 @@
+'use strict'
+
+const last = require('it-last')
+
+module.exports = ({ addAll }) => {
+  return async function add (source, options) { // eslint-disable-line require-await
+    return last(addAll(source, options))
```

Comment:
I think just passing `source` through isn't a best option here. I would much rather see `addAll([source])` all the single file inputs should work and passing non singular inputs should error anyway. Am I overlooking something here ?

Better yet would be to throw an error if multi-file input is passed as:

1. It can explain clearly that `addAll` should be used in place of `add`.
2. Error will occur before work is done as opposed to doing a work and then producing unclear error saying result isn't iterable.

If `Iterable<Uint8Array>` and `AsyncIterable<Uint8Array>` as suggested in other comment, it would be both trivial to implement and remove all the ambiguity.

Reply:
Node streams throw a spanner in the works again.  They are `AsyncIterator<Uint8Array>`, so `addAll([source])` becomes `addAll(Iterable<AsyncIterable<Uint8Array>>)` which is not something we currently support.  We could, I suppose, but so far no-one has asked for it.

I agree that `add` should reject multiple items, but this can be done in a follow up PR.